src/pages/select-metric/select-metric.ts

```diff
@@ -0,0 +1,44 @@
+import { Component } from '@angular/core';
+
+import { NavController, NavParams } from 'ionic-angular';
+
+import { BeeminderApi } from '../../providers/beeminder-api';
+
+import { CreateGoalSettingsPage } from '../create-goal-settings/create-goal-settings';
+
+@Component({
+  selector: 'page-select-metric',
+  templateUrl: 'select-metric.html'
+})
+
+export class SelectMetricPage {
+  integration: string;
+  metric: string;
+  metrics: string[];
+
+  constructor(public navCtrl: NavController, private params: NavParams, public beeminder: BeeminderApi) {
+    // this.integration = this.params.get('integration');
+
+    // dummy variable till Integration from Goal Wizard is figured out
+    this.integration = "Facebook";
+    this.metrics = this.getIntegrationMetrics();
+  }
+
+  getIntegrationMetrics() {
+    /* assuming that this function from Beeminder API returns a string array of metrics
+    return this.beeminder.getMetrics(this.integration);
+     */
+    return ['Posts Per Day', 'Likes Per Day', 'Login Per Week'];
+  }
+  selectMetric(metric) {
+    this.metric = metric;
+    this.goToCreateGoal();
```

Comment:
Yeah that is a good rule of thumb to go by, but you need to be careful with that. This is a mix of thinking too far ahead, when later on this could easily be refactored into a new function. Also if you were to debug this; calling another function will make you lose context, making you have to follow hop into between the function stacks to understand what is happening.

-Sorry for the long text :confounded: